### PR TITLE
Updates to support GUI updating in cobra-tools

### PR DESCRIPTION
**Pull request by toxicantidote**
Supports [pull request #46 to cobra-tools](https://github.com/OpenNaja/cobra-tools/pull/46) by adding support for callbacks to OVL format handler

### CODE REVIEW

File: formats/ovl/__init__.py [hunk 1]
```diff
@@ -144,8 +157,22 @@ def read_z_str(self, stream, pos):
...
+		## dummy (black hole) callback for if we decide we don't want one
+		def dummy_callback(self, *args, **kwargs):
+			return
+		
+		def print_and_callback(self, message, value = None, max = None):
+			## don't print the message if it is identical to the last one - it
+			## will slow down massively repetitive tasks
+			if self.last_print != message:
+				print(message)
+				self.last_print = message
```
Comment (toxicantidote) [inline formats/ovl/__init__.py:170]: This should give a slight speed gain where the exact same message is pumped out a bunch of times by only showing the first. That is of course, if you only want the first one shown.

Comment (HENDRIX-ZT2) [reply formats/ovl/__init__.py:170]: Looks good!

File: formats/ovl/__init__.py [hunk 2]
```diff
@@ -186,6 +213,7 @@ def read(self, stream, verbose=0, file="", commands=[], ):
 			
 			# add extensions to hash dict
 			for mime_entry in self.header.mimes:
+				self.print_and_callback("Adding extensions to hash dict", value = self.header.mimes.index(mime_entry), max = len(self.header.mimes))
```
Comment (HENDRIX-ZT2) [inline formats/ovl/__init__.py:216]: For all of those, don't use list.index() (way too costly!) but wrap list in enumerate() generator:
```
for mime_i, mime_entry in enumerate(self.header.mimes):
     ...
```

Comment (toxicantidote) [reply formats/ovl/__init__.py:216]: That's true. I should have figured you were using enumerate for a reason. I'll look at a more proper way of doing this.

As a precaution, the callback used (in cobra-tools) won't issue any GUI updates if nothing has actually changed.

Comment (toxicantidote) [reply formats/ovl/__init__.py]: I've now pushed in another commit that calls len() once at the start, then simply increments a variable to count through.

You're absolutely right - removing the repeated calls to .index() has sped things up noticeably.

Comment (toxicantidote) [reply formats/ovl/__init__.py]: I'll push through something similar for _cobra-tools_ momentarily.

**Pull request merged**